modules/_labs/pagination/react/lib/Pages.tsx

```diff
@@ -0,0 +1,138 @@
+import styled from '@emotion/styled-base';
+import {IconButton} from '@workday/canvas-kit-react-button';
+import canvas, {borderRadius} from '@workday/canvas-kit-react-core';
+import _ from 'lodash';
+import React from 'react';
+
+const ActivePage = styled('button')({
+  display: 'inline-flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  fontSize: 13,
+  fontWeight: 700,
+  height: 32,
+  width: 32,
+  backgroundColor: canvas.colors.blueberry400,
+  color: canvas.colors.frenchVanilla100,
+  borderRadius: borderRadius.m,
+  ':focus': {outline: 'none'},
+});
+
+const NoPointerEvents = styled('div')({
+  pointerEvents: 'none',
+});
+
+const Ellipse = () => (
+  <NoPointerEvents>
+    <IconButton
+      key={'ellipse'}
+      aria-label={`Navigation Ellipse`}
+      variant={IconButton.Variant.Square}
+      size={IconButton.Size.Small}
+      tabIndex={-1}
+    >
+      ...
+    </IconButton>
+  </NoPointerEvents>
+);
+
+const Pages: React.FC<{
+  numPages: number;
+  currentPage: number;
+  clickHandler: (page: number) => void;
+}> = props => {
+  const {numPages, currentPage, clickHandler} = props;
+  const [width, setWidth] = React.useState(window.innerWidth);
+
+  React.useEffect(() => {
+    const handleWindowSizeChange = () => setWidth(window.innerWidth);
+    window.addEventListener('resize', handleWindowSizeChange);
+    return () => window.removeEventListener('resize', handleWindowSizeChange);
+  }, []);
+  const mobile = width < 500;
+
+  let pagesToDisplay = 5;
+  let start = 1;
+
+  if (mobile) {
+    start = currentPage;
+    if (currentPage >= numPages - 1) {
+      start = Math.max(numPages - 2, 1);
+      pagesToDisplay = 3;
+    } else {
+      pagesToDisplay = 1;
+    }
+  } else {
+    const midwayAndMorePages = currentPage > 2 && numPages > pagesToDisplay;
+    if (midwayAndMorePages) {
+      const oneOfLastThreePages = currentPage >= numPages - 3;
+      if (oneOfLastThreePages) {
+        start = numPages - 4;
+      } else {
+        start = currentPage - 2;
+      }
+    }
+  }
+
+  const end = Math.min(start + pagesToDisplay, numPages + 1);
+  const pages = _.range(start, end);
+
+  const PaginationButton: React.FC<{
+    page: number;
+  }> = props => (
+    <IconButton
+      data-testid={`paginationButton${props.page}`}
```

Comment:
Should we have these data-testids hard-coded in the components? The other idea is automation helpers since DOM is usually an implementation detail of components.

Reply:
@NicholasBoll I went ahead and remove all usages of data-testid